### PR TITLE
Add tests that check removed or renamed classes are removed from internal cache

### PR DESCRIPTION
**Pull request by zaki50**
This PR required for #5037 to be merged first.

There seems to be a bug in core.

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java [hunk 1]
```diff
@@ -322,4 +324,43 @@ public void remove_shouldRemoveInfoFromPKTable() {
...
+
+    @Test
+    public void rename_then_cancel() {
+        if (type == SchemaType.IMMUTABLE) {
+            return;
+        }
+
+        realmSchema.create("foo");
+        // getSchemaForClass is an internal method, but used from DynamicRealmObject and RealmQuery
+        final RealmObjectSchema foo = realmSchema.getSchemaForClass("foo");
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java]: Wouldn't using the public `realmSchema.get()` be the same?

File: realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java [hunk 2]
```diff
@@ -322,4 +324,43 @@ public void remove_shouldRemoveInfoFromPKTable() {
...
+
+    @Test
+    public void remove_then_cancel() {
+        if (type == SchemaType.IMMUTABLE) {
+            return;
+        }
+
+        realmSchema.create("foo");
+        // getSchemaForClass is an internal method, but used from DynamicRealmObject and RealmQuery
+        realmSchema.getSchemaForClass("foo"); // make cache entry
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java]: Wouldn't using the public realmSchema.get() be the same?

Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java]: No. `realmSchema.get()` is not using the cache. `getSchemaForClass(Class)` also has the same issue, but that method is only used for typed models and no classes are removed actually.

File: realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java (hunk 1 above)
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java]: Same

**Pull request closed without merging**